app/src/main/java/com/keylesspalace/tusky/db/TimelineDao.kt

```diff
@@ -86,18 +87,39 @@ WHERE timelineUserId = :accountId AND (serverId = :statusId OR reblogServerId =
     )
     abstract fun removeAllByUser(accountId: Long, userId: String)
 
+    /**
+     * Removes everything in the TimelineStatusEntity and TimelineAccountEntity tables for one user account
+     * @param accountId id of the account for which to clean tables
+     */
+    @Transaction
+    open suspend fun removeAll(accountId: Long) {
+        removeAllStatuses(accountId)
+        removeAllAccounts(accountId)
+    }
+
     @Query("DELETE FROM TimelineStatusEntity WHERE timelineUserId = :accountId")
-    abstract fun removeAllForAccount(accountId: Long)
+    abstract suspend fun removeAllStatuses(accountId: Long)
 
     @Query("DELETE FROM TimelineAccountEntity WHERE timelineUserId = :accountId")
-    abstract fun removeAllUsersForAccount(accountId: Long)
+    abstract suspend fun removeAllAccounts(accountId: Long)
 
     @Query(
         """DELETE FROM TimelineStatusEntity WHERE timelineUserId = :accountId
 AND serverId = :statusId"""
     )
     abstract fun delete(accountId: Long, statusId: String)
 
+    /**
+     * Cleans the TimelineStatusEntity and TimelineAccountEntity tables from old entries.
+     * @param accountId id of the account for which to clean tables
+     * @param limit how many statuses to keep
+     */
+    @Transaction
```

Comment:
Same goes for this transaction. I don't think we really need transaction guarantees here